apollo-router/src/query_planner/convert.rs

```diff
@@ -195,18 +190,15 @@ impl From<&'_ next::DeferredDeferBlock> for plan::DeferredNode {
             query_path: crate::json_ext::Path(
                 query_path
                     .iter()
-                    .filter_map(|e| match e {
-                        next::QueryPathElement::Field(field) => Some(
-                            // TODO: type conditioned fetching once it s available in the rust planner
-                            crate::graphql::JsonPathElement::Key(
-                                field.response_key().to_string(),
-                                None,
-                            ),
-                        ),
-                        next::QueryPathElement::InlineFragment(inline) => {
-                            inline.type_condition.as_ref().map(|cond| {
-                                crate::graphql::JsonPathElement::Fragment(cond.to_string())
-                            })
+                    .map(|e| match e {
+                        next::QueryPathElement::Field { response_key } =>
+                        // TODO: type conditioned fetching once it s available in the rust planner
+                        {
+                            crate::graphql::JsonPathElement::Key(response_key.to_string(), None)
+                        }
```

Comment:
Unrelated to this PR: looks like we are not handling type conditioned here, and probably should be.

Reply:
Filed https://apollographql.atlassian.net/browse/ROUTER-1143 for that.